common/djangoapps/third_party_auth/utils.py

```diff
@@ -57,3 +76,19 @@ def validate_uuid4_string(uuid_string):
     except ValueError:
         return False
     return True
+
+
+def is_saml_provider(backend, kwargs):
+    """ Verify that the third party provider uses SAML """
+    current_provider = provider.Registry.get_from_pipeline({'backend': backend, 'kwargs': kwargs})
```

Comment:
@zamanafzal This will always return `None`.

https://github.com/edx/edx-platform/blob/a3f5169bf848ab8c3b3575ac2218156464eb43fc/common/djangoapps/third_party_auth/provider.py#L85-L98

If you check the code above, it's using `is_active_for_pipeline`, which is shown below.

https://github.com/edx/edx-platform/blob/a3f5169bf848ab8c3b3575ac2218156464eb43fc/common/djangoapps/third_party_auth/models.py#L701-L703

The backend name of the provider (`self.backend_name`) is being compared with the pipeline's backend (`pipeline['backend']`). So the backend passed to the pipeline should be the backend's name, a string value and not the backend class that is being passed.

https://github.com/edx/edx-platform/blob/a3f5169bf848ab8c3b3575ac2218156464eb43fc/common/djangoapps/third_party_auth/pipeline.py#L787

---

Therefore, it should be:
```python
    current_provider = provider.Registry.get_from_pipeline({'backend': backend.name, 'kwargs': kwargs})
```

I ended up closing a pull request i created for this, but, for future reference, it was https://github.com/edx/edx-platform/pull/27079.